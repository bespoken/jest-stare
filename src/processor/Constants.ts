import * as chalk from "chalk";

/**
 * Contains Constants constants
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Name (same as package.json)
     * @static
     * @memberof Constants
     */
    public static readonly NAME = "jest-stare";

    /**
     * Logo for messaging
     * @static
     * @memberof Constants
     */
    public static readonly LOGO = chalk.default.green("**  ") + chalk.default.green("jest") + chalk.default.yellow("-") + chalk.default.red("stare");

    /**
     * Suffix for logo
     * @static
     * @memberof Constants
     */
    public static readonly SUFFIX = chalk.default.green("\t**");

    /**
     * Default jest-stare results directory
     * @static
     * @memberof Constants
     */
    public static readonly DEFAULT_RESULTS_DIR = "./" + Constants.NAME;

    /**
     * File extension for HTML files
     * @static
     * @memberof Constants
     */
    public static readonly HTML_EXTENSION = ".html";


    /**
     * Main HTML index file for report
     * @static
     * @memberof Constants
     */
    public static readonly MAIN_HTML = "index" + Constants.HTML_EXTENSION;

    /**
     * jest-stare render JS file
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_JS = "view.js";

    /**
     * Message for reporter run of jest-stare
     * @static
     * @memberof Constants
     */
    public static readonly REPORTER_WRITTING = " will write each completed run to ";

    /**
     * Raw JSON response results
     * @static
     * @memberof Constants
     */
    public static readonly RESULTS_RAW = "jest-results.json";

    /**
     * jest-stare custom css
     * @static
     * @memberof Constants
     */
    public static readonly JEST_STARE_CSS = Constants.NAME + ".css";

    /**
     * Template HTML file
     * @static
     * @memberof Constants
     */
    public static readonly TEMPLATE_HTML = "template.html";

    /**
     * Template HTML file
     * @static
     * @memberof Constants
     */
    public static readonly TEMPLATE_INLINE_SOURCE_HTML = "templateInlineSource.html";

    /**
     * Base dir to contain css files
     * @static
     * @memberof Constants
     */
    public static readonly CSS_DIR = "css/";

    /**
     * Base dir to contain js files
     * @static
     * @memberof Constants
     */
    public static readonly JS_DIR = "js/";

    /**
     * Base dir to contain image snapshot diff files
     * @static
     * @memberof Constants
     */
    public static readonly IMAGE_SNAPSHOT_DIFF_DIR = "image_snapshot_diff/";

    /**
     * jest option for test results processors
     * @static
     * @memberof Constants
     */
    public static readonly TEST_RESULTS_PROCESSOR = "--testResultsProcessor";

    /**
     * jest option for reporters
     * @static
     * @memberof Constants
     */
    public static readonly REPORTERS = "--reporters";

    /**
     * Log content after logo
     * @static
     * @memberof Constants
     */
    public static readonly LOG_MESSAGE = ": wrote output report to ";

    /**
     * Merging content message
     * @static
     * @memberof Constants
     */
    public static readonly MERGE_MESSAGE = ": will merge with ";

    /**
     * Error for whenever called without test results
     * @static
     * @memberof Constants
     */
    public static readonly NO_INPUT = Constants.NAME + " was called without input results";

    /**
     * Error for whenever the CLI is called without an input file is specified containing jest JSON
     * @static
     * @memberof Constants
     */
    public static readonly NO_CLI_INPUT = Constants.NAME + " CLI was called without input JSON file to read";

    /**
     * Notification called with programmatic config
     * @static
     * @memberof Constants
     */
    public static readonly OVERRIDE_JEST_STARE_CONFIG = Constants.NAME + " was called with programmatic config";
}
