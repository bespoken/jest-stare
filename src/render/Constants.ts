/**
 * Constants for rendering
 * @export
 * @class Constants
 */
export class Constants {

    /**
     * Label for passed tests
     * @static
     * @memberof Constants
     */
    public static readonly PASSED_LABEL = "Passed";

    /**
     * Label for failed tests
     * @static
     * @memberof Constants
     */
    public static readonly FAILED_LABEL = "Failed";

    /**
     * Label for pending tests
     * @static
     * @memberof Constants
     */
    public static readonly PENDING_LABEL = "Pending";

    /**
     * Label for obsolete info
     * @static
     * @memberof Constants
     */
    public static readonly OBSOLETE_SNAPSHOT_TEST_LABEL = "Obsolete Test";

    /**
     * Label for changed info
     * @static
     * @memberof Constants
     */
    public static readonly CHANGED_LABEL = "Changed";

    /**
     * Label for added info
     * @static
     * @memberof Constants
     */
    public static readonly ADDED_LABEL = "Added";


    /**
     * Label for removed snapshot tests
     * @static
     * @memberof Constants
     */
    public static readonly UPDATED_SNAPSHOT_TEST_LABEL = "Updated Snapshot Test";

    /**
     * Label for removed snapshot files
     * @static
     * @memberof Constants
     */
    public static readonly REMOVED_OBSOLETE_SNAPSHOT_FILE_LABEL = "Removed Obsolete Snapshot File";

    /**
     * Label for removed info
     * @static
     * @memberof Constants
     */
    public static readonly OBSOLETE_SNAPSHOT_FILE_LABEL = "Obsolete File";

    /**
     * Passed jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_PASS = "passed";

    /**
     * Failed jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_FAIL = "failed";

    /**
     * Pending jest status
     * @static
     * @memberof Constants
     */
    public static readonly TEST_STATUS_PEND = "pending";

    /**
     * Obsolete snapshot file
     * @static
     * @memberof Constants
     */
    public static readonly OBSOLETE_SNAPSHOT_FILE = "#f8f9fa"; // btn-light

    /**
     * Obsolete color
     * @static
     * @memberof Constants
     */
    public static readonly OBSOLETE_SNAPSHOT_TEST = "#ffc107"; // btn-warning

    /**
     * Added color
     * @static
     * @memberof Constants
     */
    public static readonly ADDED = "#007bff"; // btn-primary

    /**
     * Removed snapshot test color
     * @static
     * @memberof Constants
     */
    public static readonly UPDATED_SNAPSHOT_TEST = "#17a2b8"; // btn-info

    /**
     * Removed snapshot file color
     * @static
     * @memberof Constants
     */
    public static readonly REMOVED_OBSOLETE_SNAPSHOT_FILE = "#343a40"; // btn-dark

    /**
     * Changed color
     * @static
     * @memberof Constants
     */
    public static readonly CHANGED = "#6c757d"; // btn-secondary

    /**
     * Pass color
     * @static
     * @memberof Constants
     */
    public static readonly PASS_RAW = "28a745"; // btn-success
    public static readonly PASS = "#" + Constants.PASS_RAW;

    /**
     * Fail color
     * @static
     * @memberof Constants
     */
    public static readonly FAIL_RAW = "dc3545"; // btn-danger
    public static readonly FAIL = "#" + Constants.FAIL_RAW;

    /**
     * Pending color
     * @static
     * @memberof Constants
     */
    public static readonly PENDING_RAW = "ffc107";
    public static readonly PENDING = "#" + Constants.PENDING_RAW; // btn-warning

    /**
     * Passed test class
     * @static
     * @memberof Constants
     */
    public static readonly PASSED_TEST = "passed-test";

    /**
     * Failed test class
     * @static
     * @memberof Constants
     */
    public static readonly FAILED_TEST = "failed-test";

    /**
     * Pending test class
     * @static
     * @memberof Constants
     */
    public static readonly PENDING_TEST = "pending-test";
}
