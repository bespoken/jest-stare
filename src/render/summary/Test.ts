import { Constants } from "../Constants";

/**
 * Create elements for a test
 * @export
 * @class Test
 */
export class Test {

    /**
     * Add test to a section of the presented table
     * @static
     * @param {IInnerTestResults} innerTestResult - see object description
     * @param {HTMLDivElement} element - base element to add to
     * @returns {HTMLDivElement} - populated element
     * @memberof Test
     */
    public static create(innerTestResult: jest.AssertionResult): HTMLElement {
        const containerDiv = document.createElement("div") as HTMLDivElement;

        const anchor = document.createElement("a") as HTMLAnchorElement;
        anchor.href = "#" + innerTestResult.title.replace(/\s+/g, "-").toLowerCase();

        const testName = document.createElement("span") as HTMLElement;
        testName.textContent = innerTestResult.title;

        anchor.appendChild(Test.getSimbolSpanFromStatus(innerTestResult.status));
        anchor.appendChild(testName);

        containerDiv.appendChild(anchor);

        return containerDiv;
    }

    /**
     * Generates a span with a symbol and appropiate color from the test status
     * @static
     * @param {status} string - the test status
     * @returns {HTMLSpanElement} - populated element
     * @memberof Test
     */
    private static  getSimbolSpanFromStatus(status: string) {
        const span = document.createElement("span") as HTMLSpanElement;
        span.classList.add("summary-test-label", "test");

        if (status === Constants.TEST_STATUS_PASS) {
            span.textContent = "âœ“";
            span.classList.add("pass");
        }

        if (status === Constants.TEST_STATUS_PEND) {
            span.textContent = "O";
            span.classList.add("pending");
        }

        if (status === Constants.TEST_STATUS_FAIL) {
            span.textContent = "X";
            span.classList.add("fail");
        }

        return span;
    }
}
