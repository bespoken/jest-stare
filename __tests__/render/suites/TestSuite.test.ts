jest.mock("../../../src/render/tests/Test");

import { TestSuite } from "../../../src/render/suites/TestSuite";
import { Test } from "../../../src/render/tests/Test";
import { Constants } from "../../../src/render/Constants";

const simplePassingTests: jest.AggregatedResult = require("../../__resources__/simplePassingTests.json");
const simpleFailingTests: jest.AggregatedResult = require("../../__resources__/simpleFailingTests.json");
const twoSideBySideFailDiffTests: jest.AggregatedResult = require("../../__resources__/twoSideBySideFailDiffTests.json");
// const largeTests: jest.AggregatedResult = require("../../__resources__/largeTests.json");

// mock out the individual tests within describe blocks
(Test.create as any) = jest.fn( () => {
    const div = document.createElement("div") as HTMLDivElement;
    div.textContent = "dummy div";
    return div;
});

const createTestResult = (status: jest.Status) => {
        return {
            status,
            ancestorTitles: ["ancestor"],
            failureMessages: [],
            fullName: "",
            numPassingAsserts: 0,
            title: "",
        };
};

describe("TestSuite tests", () => {
    describe("Snapshots", () => {
        it("should create proper elements for simple passing tests", () => {
            expect(TestSuite.create(simplePassingTests)).toMatchSnapshot();
        });

        it("should create proper elements for simple failing tests", () => {
            expect(TestSuite.create(simpleFailingTests)).toMatchSnapshot();
        });

        it("should create proper elements for two diffs in html", () => {
            expect(TestSuite.create(twoSideBySideFailDiffTests)).toMatchSnapshot();
        });

        it("should create proper elements for large test > 500 tests", () => {
            expect(TestSuite.create(twoSideBySideFailDiffTests)).toMatchSnapshot();
        });
    });

    it("should assign test status correctly when all are the same", () => {
        let testStatusClass;

        const testSectionStatus: Map<string, string> = new Map<string, string>();

        const testResult =  {
            testResults: [
                createTestResult("passed"),
                createTestResult("passed"),
            ]
        };

        for (const result of testResult.testResults) {
            testStatusClass = TestSuite.asignStatus(testStatusClass, result, testSectionStatus);
        }

        expect(testStatusClass).toEqual(Constants.PASSED_TEST);
        expect(testSectionStatus.get("ancestor")).toEqual(Constants.PASSED_TEST);

    });

    it("should assign test status correctly when are different", () => {
        let testStatusClass;

        const testSectionStatus: Map<string, string> = new Map<string, string>();

        const testResult =  {
            testResults: [
                createTestResult("passed"),
                createTestResult("failed"),
            ]
        };

        for (const result of testResult.testResults) {
            testStatusClass = TestSuite.asignStatus(testStatusClass, result, testSectionStatus);
        }

        expect(testStatusClass).toEqual(Constants.FAILED_TEST + TestSuite.JOIN_CHAR + Constants.PASSED_TEST);
        expect(testSectionStatus.get("ancestor")).toEqual(Constants.FAILED_TEST + TestSuite.JOIN_CHAR + Constants.PASSED_TEST);

    });
});
