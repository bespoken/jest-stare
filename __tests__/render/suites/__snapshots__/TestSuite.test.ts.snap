// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite tests should create proper elements for large test > 500 tests 1`] = `
Array [
  <div
    class="my-3 p-3 bg-white rounded box-shadow both-test"
    id="jest-stare\\\\__tests__\\\\example\\\\FailingTests.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\example\\FailingTests.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow both-test"
      id="Failing Tests.0.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Failing Tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\render\\\\Render.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\render\\Render.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Render tests.1.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Render tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\processor\\\\Processor.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\processor\\Processor.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Processor tests.2.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Processor tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\utils\\\\IO.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\utils\\IO.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="IO tests.3.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        IO tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\utils\\\\Logger.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\utils\\Logger.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Logger tests.4.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Logger tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\reporter\\\\Reporter.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\reporter\\Reporter.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Reporter tests.5.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Reporter tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
]
`;

exports[`TestSuite tests should create proper elements for simple failing tests 1`] = `
Array [
  <div
    class="my-3 p-3 bg-white rounded box-shadow both-test"
    id="Work\\\\Dev\\\\node\\\\jest-stare\\\\__tests__\\\\utils\\\\IO.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      Work\\Dev\\node\\jest-stare\\__tests__\\utils\\IO.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow both-test"
      id="IO tests.0.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        IO tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="Work\\\\Dev\\\\node\\\\jest-stare\\\\__tests__\\\\reporter\\\\Reporter.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      Work\\Dev\\node\\jest-stare\\__tests__\\reporter\\Reporter.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Reporter tests.1.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Reporter tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
]
`;

exports[`TestSuite tests should create proper elements for simple passing tests 1`] = `
Array [
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="Work\\\\Dev\\\\node\\\\jest-stare\\\\__tests__\\\\utils\\\\IO.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      Work\\Dev\\node\\jest-stare\\__tests__\\utils\\IO.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="IO tests.0.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        IO tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="Work\\\\Dev\\\\node\\\\jest-stare\\\\__tests__\\\\reporter\\\\Reporter.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      Work\\Dev\\node\\jest-stare\\__tests__\\reporter\\Reporter.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Reporter tests.1.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Reporter tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
]
`;

exports[`TestSuite tests should create proper elements for two diffs in html 1`] = `
Array [
  <div
    class="my-3 p-3 bg-white rounded box-shadow both-test"
    id="jest-stare\\\\__tests__\\\\example\\\\FailingTests.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\example\\FailingTests.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow both-test"
      id="Failing Tests.0.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Failing Tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\render\\\\Render.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\render\\Render.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Render tests.1.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Render tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\processor\\\\Processor.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\processor\\Processor.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Processor tests.2.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Processor tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\utils\\\\IO.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\utils\\IO.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="IO tests.3.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        IO tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\utils\\\\Logger.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\utils\\Logger.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Logger tests.4.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Logger tests
      </h6>
      <div>
        dummy div
      </div>
    </div>
  </div>,
  <div
    class="my-3 p-3 bg-white rounded box-shadow passed-test"
    id="jest-stare\\\\__tests__\\\\src\\\\reporter\\\\Reporter.test.ts"
  >
    <h5
      class="border-bottom pb-2 mb-0 display-5"
    >
      jest-stare\\__tests__\\src\\reporter\\Reporter.test.ts
    </h5>
    <div
      class="my-3 p-3 bg-white rounded box-shadow passed-test"
      id="Reporter tests.5.0.0"
    >
      <h6
        class="border-bottom pb-2 mb-0 display-6"
      >
        Reporter tests
      </h6>
      <div>
        dummy div
      </div>
      <div>
        dummy div
      </div>
    </div>
  </div>,
]
`;
